framework/src/modules/chain/chain.js

```diff
@@ -158,6 +163,10 @@ module.exports = class Chain {
 			// Ready to bind modules
 			scope.logic.peers.bindModules(scope.modules);
 
+			this.channel.subscribe('lisk:state:updated', event => {
+				Object.assign(scope.applicationState, event.data);
```

Comment:
prefer not to use `Object.assign`

Reply:
Also, for applicationState, it should update the whole object

Reply:
We decide to keep `Object.assign` for this specific use case.